scripts/ssl-manager.js

```diff
@@ -577,6 +585,72 @@ function SSLManager(config) {
         return { result: 0 };
     };
 
+    me.configureNodeMemory = function configureNodeMemory() {
+        var resp;
+
+        resp = me.defineNodeMemory();
+        if (resp.result != 0) return resp;
+
+        if (config.nodeMemory >= REQUIRED_MEM) {
+            return me.exec(me.setClouletsValidation);
+        } else {
+            return error(Response.ERROR_UNKNOWN, "At least 512 MB RAM (4 cloudlets) are recommended for the correct installation of the Let's Encrypt add-on.");
+        }
+    };
+
+    me.defineNodeMemory = function() {
+        var resp;
+
+        resp = me.cmd("free -m | grep Mem | awk '{print $2}'", {
+            nodeGroup: config.nodeGroup
+        });
+        config.nodeMemory = Number(resp.responses[0].out);
+        return {result: 0};
+    };
+
+    me.removeNodeValidation = function removeNodeValidation() {
+        var nodeGroupValidations,
+            platformVersion = getPlatformVersion();
+
+        nodeGroupValidations = nodeManager.getNodeGroupValidations();
+
+        if (config.setValidations && nodeGroupValidations) {
+            nodeGroupValidations.minCloudlets = "";
+
+            if (compareVersions(platformVersion, '5.8') >= 0) {
+                return jelastic.env.control.ApplyNodeGroupData(config.envName, session, config.nodeGroup, {"validation": nodeGroupValidations});
+            }
+        }
+
+        return { result: 0 };
+    };
+
+    me.setClouletsValidation = function() {
+        var nodeGroupValidations,
+            platformVersion,
+            cloudletsAmount,
+            resp;
+
+        nodeGroupValidations = nodeManager.getNodeGroupValidations() || {};
+
+        if (!nodeGroupValidations.minCloudlets) {
+            cloudletsAmount = parseInt(REQUIRED_MEM / me.getCloudletsMemAmount());
+            nodeGroupValidations.minCloudlets = cloudletsAmount;
+            platformVersion = getPlatformVersion();
+
+            if (compareVersions(platformVersion, '5.8') >= 0) {
+                resp = jelastic.env.control.ApplyNodeGroupData(config.envName, session, config.nodeGroup, {"validation": nodeGroupValidations});
+                if (resp.result != 0) return resp;
+                config.setValidations = true;
+            }
+        }
+        return { result: 0 };
+    };
+
+    me.getCloudletsMemAmount = function() {
+        return CLOUDLET_MEM_AMOUNT; // TODO: read from system settings
```

Comment:
must be calculated: nodeMemory / (max(envInfo.node.fixedCloudlets, envInfo.node.flexibleCloudlets))

Reply:
done